### PR TITLE
feat: add directConnect feature

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Change list

Add https://github.com/appium/java-client/pull/1747 for dotnet client.

Copy from the ticket:

---

I'd like to add https://appiumpro.com/editions/86-connecting-directly-to-appium-hosts-in-distributed-environments for Java client. Other clients such as webdriverio, Ruby and Python already have them for long. If the response had directConnect capabilities, the appium client attempt to send requests to the directConnect one instead of the original URL.

- directConnectProtocol
- directConnectHost
- directConnectPort
- directConnectPath

or `appium:` prefixed ones.

---

I saw a ticekt in selenium repo that they are going to add ClientConfig (currently Java client has), so as same as https://github.com/appium/java-client/pull/1747, I have added `AppiumClientConfig` to keep client side config so that we'll be able to inherit the selenium project's one when they introduce it. Then, we will not need to add a new constructor.

(With this, all of our clients and webdrierio support this.)

## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] New feature (non-breaking change which adds value to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Documentation
- [ ] Have you proposed a file change/ PR with appium to update documentation? 
## Integration tests
- [ ] Have you provided integration tests to pass against the beta version of appium? (for Bugfix or New feature)

## Details

Please provide more details about changes if it is necessary. If there are new features you can provide code samples which show the way they
work and possible use cases. Also you can create [gists](https://gist.github.com) with pasted C# code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://help.github.com/categories/writing-on-github/)

### CODE REVIEW

File: src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs
Comment (Dor-bl) [reply src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs:70]: see https://github.com/appium/dotnet-client/pull/629

Comment (KazuCocoa) [reply src/Appium.Net/Appium/Service/AppiumCommandExecutor.cs:70]: ah, the refactoring seems good idea to me

**Review by Dor-bl — APPROVED** [review]

**Comment by Dor-bl** [comment]
@KazuCocoa LGTM, After you merge I'll rebase and update my PR for `public Response Execute(Command commandToExecute)`

**Comment by KazuCocoa** [comment]
thank you!

File: src/Appium.Net/Appium/AppiumDriver.cs
Comment (cwood-panopto) [inline src/Appium.Net/Appium/AppiumDriver.cs:78]: @KazuCocoa This seems to have broken being able to specify your own command timeout in 5.0.0-rc.1, which pretty much breaks Saucelabs because their real devices can take longer than 60 seconds to be provisioned.

Comment (Dor-bl) [reply src/Appium.Net/Appium/AppiumDriver.cs:78]: I already fixed this one on another PR. I need to release a new version

**Pull request merged**
